test/pipe.cc

```diff
@@ -0,0 +1,96 @@
+#include "eventuals/pipe.h"
+
+#include <thread>
+
+#include "eventuals/loop.h"
+#include "eventuals/terminal.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+
+using eventuals::Loop;
+using eventuals::Pipe;
+
+TEST(Pipe, UniqueValue) {
+  Pipe<int> pipe;
+
+  *pipe.Write(1);
+  auto e = [&pipe]() {
+    return pipe.Read()
+        | Loop<int>()
+              .context(/*sum = 0*/ 0)
+              .body([](auto&& sum, auto& repeated, auto&& value) {
+                sum += value;
+                repeated.Done();
+              })
+              .ended([](auto& sum, auto& k) {
+                k.Start(std::move(sum));
+              });
+  };
+
+  EXPECT_EQ(1, pipe.values_size());
```

Comment:
Don't worry about checking this, just make sure all the values you wrote on the pipe were read on the pipe.

Reply:
I think it's fine to have a `Size()` function, it's not adding anything extra in the tests. But you should create an separate test just for it since you added it please!